pkg/reconciler/service/reconciler.go

```diff
@@ -326,7 +328,12 @@ func (r *ComponentReconciler) newRouter(ctx context.Context, workerPool *ants.Po
 			}
 
 			//enrich logger with correlation ID and component name
-			r.logger = r.logger.With(zap.Field{Key: "correlation-id", Type: zapcore.StringType, String: model.CorrelationID}, zap.Field{Key: "component-name", Type: zapcore.StringType, String: model.Component})
+			loggerNew, err := logger.NewLogger(r.debug)
+			if err != nil {
+				r.logger.Errorf("Could not create new logger which is correlationID aware: %s", err)
```

Comment:
ambiguous. My best guess:

```suggestion
				r.logger.Errorf("Could not create a new logger that is correlationID-aware: %s", err)
```

Pls confirm this is the correct interpretation

Reply:
Yes, definitely correct :)